### DISCUSSION TITLE
Can't update to 18.04.1?

### DISCUSSION THREAD

**Discussion by Mario0051** [microsoft/WSL#3489] [General]
* Windows version / build number:
```
Microsoft Windows [Version 10.0.18219.1000]
```
* Terminal output:
```
:~$ sudo -S env RELEASE_UPGRADER_NO_SCREEN=1 do-release-upgrade
Checking for a new Ubuntu release
Get:1 Upgrade tool signature [819 B]
Get:2 Upgrade tool [1,258 kB]
Fetched 1,258 kB in 0s (0 B/s)
authenticate 'bionic.tar.gz' against 'bionic.tar.gz.gpg'
extracting 'bionic.tar.gz'

Reading cache
pcilib: Cannot open /proc/bus/pci
lspci: Cannot find any working access method.

Checking package manager
Reading package lists... Done
Building dependency tree
Reading state information... Done
Hit http://security.ubuntu.com/ubuntu xenial-security InRelease
Hit http://archive.ubuntu.com/ubuntu xenial InRelease
Hit http://archive.ubuntu.com/ubuntu xenial-updates InRelease
Hit http://archive.ubuntu.com/ubuntu xenial-backports InRelease
Fetched 0 B in 6s (0 B/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done

Restoring original system state

Aborting
Reading package lists... Done
Building dependency tree
Reading state information... Done
```
* Expected behavior: Update correctly without ```Restoring original system state```.

**Comment by rbalint** [comment]
Have you run `sudo apt-get update && sudo apt-get upgrade` before trying to upgrade?
`do-release-upgrade` is passing for me in WSL in a fully upgraded Xenial installation.

**Comment by rbalint** [comment]
Could you please try again and if the upgrade fails check if there is useful information in the logs in `/var/log/dist-upgrade/` ?

**Comment by Mario0051** [comment]
I tried that and in the main.log file it says:
```
2018-08-21 12:05:59,242 INFO Using config files '['./DistUpgrade.cfg.xenial']'
2018-08-21 12:05:59,242 INFO uname information: 'Linux DESKTOP-T2D0OGA 4.4.0-18219-Microsoft #1000-Microsoft Fri Aug 10 17:21:00 PST 2018 x86_64'
2018-08-21 12:05:59,534 INFO apt version: '1.2.27'
2018-08-21 12:05:59,534 INFO python version: '3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609]'
2018-08-21 12:05:59,552 INFO release-upgrader version '18.04.24' started
2018-08-21 12:05:59,744 INFO locale: 'en_US' 'UTF-8'
2018-08-21 12:06:00,076 WARNING estimate_kernel_initrd_size_in_boot() returned '0' for kernel?
2018-08-21 12:06:00,077 WARNING estimate_kernel_initrd_size_in_boot() returned '0' for initrd?
2018-08-21 12:06:01,836 DEBUG Using 'DistUpgradeViewText' view
2018-08-21 12:06:02,734 DEBUG enable dpkg --force-overwrite
2018-08-21 12:06:03,795 DEBUG creating statefile: '/var/log/dist-upgrade/apt-clone_system_state.tar.gz'
2018-08-21 12:06:21,085 DEBUG lsb-release: 'xenial'
2018-08-21 12:06:21,091 DEBUG _pythonSymlinkCheck run
2018-08-21 12:06:21,094 DEBUG openCache()
2018-08-21 12:06:21,094 DEBUG No such plugin directory: ./plugins
2018-08-21 12:06:21,095 DEBUG plugins for condition 'PreCacheOpen' are '[]'
2018-08-21 12:06:21,095 DEBUG plugins for condition 'bionicPreCacheOpen' are '[]'
2018-08-21 12:06:21,095 DEBUG plugins for condition 'from_xenialPreCacheOpen' are '[]'
2018-08-21 12:06:21,095 DEBUG quirks: running PreCacheOpen
2018-08-21 12:06:21,095 DEBUG running Quirks.PreCacheOpen
2018-08-21 12:06:26,697 DEBUG /openCache(), new cache size 56931
2018-08-21 12:06:26,698 DEBUG need_server_mode(): run in 'desktop' mode, (because of pkg 'ubuntu-desktop')
2018-08-21 12:06:26,698 DEBUG checkViewDepends()
2018-08-21 12:06:26,699 DEBUG running doUpdate() (showErrors=False)
2018-08-21 12:07:20,677 DEBUG openCache()
2018-08-21 12:07:21,631 DEBUG /openCache(), new cache size 56931
2018-08-21 12:07:21,632 DEBUG doPostInitialUpdate
2018-08-21 12:07:21,632 DEBUG plugins for condition 'PostInitialUpdate' are '[]'
2018-08-21 12:07:21,632 DEBUG plugins for condition 'bionicPostInitialUpdate' are '[]'
2018-08-21 12:07:21,632 DEBUG plugins for condition 'from_xenialPostInitialUpdate' are '[]'
2018-08-21 12:07:21,632 DEBUG quirks: running bionicPostInitialUpdate
2018-08-21 12:07:21,632 DEBUG running Quirks.bionicPostInitialUpdate
2018-08-21 12:07:22,703 DEBUG abort called
2018-08-21 12:07:22,707 DEBUG openCache()
2018-08-21 12:07:22,707 DEBUG failed to SystemUnLock() (E:Not locked) 
2018-08-21 12:07:23,245 DEBUG /openCache(), new cache size 56931
```

**Comment by Mario0051** [comment]
Sometimes it hangs at ```Reading state information... Done```.

**Reply by RickyPatel** [reply]
Was facing the same issue. 
Run this command first: sudo apt-get purge snapd
Then upgrade works fine for me

**Reply by andras-varro** [reply]
This was working for me. (Upgrade 20.04->21.10)

**Reply by jonlighthall** [reply]
This worked for me for upgrading both 16.04.7 LTS and 18.04.06 LTS to 22.04.2 LTS on WSL2 and WSL1, respectively.

**Reply by RyanKruger** [reply]
> Was facing the same issue. Run this command first: sudo apt-get purge snapd Then upgrade works fine for me

This resolved my issue upgrading from Ubuntu 18.04 (bionic) to Ubuntu 20.04 (focal) on WSL1

**Comment by lewisdonofrio** [comment]
did you fix dbus?  I had to apt-get install inxi -y to get my updates working after I ran:

donofrio@localhost:~ $ cat fix-dbus.sh
echo "Implementing udev workaround (https://github.com/Microsoft/WSL/issues/143#issuecomment-209072634)" >&2
sudo tee /usr/sbin/policy-rc.d > /dev/null <<EOF
#!/bin/sh
exit 101
EOF
sudo chmod +x /usr/sbin/policy-rc.d
sudo dpkg-divert --local --rename --add /sbin/initctl
sudo ln -fs /bin/true /sbin/initctl
donofrio@localhost:~$

**Comment by Mario0051** [comment]
I tried running ```apt-get install inxi -y``` to see if it would be fixed and it still has the same problem.

**Comment by imReker** [comment]
Same problem and log as you. 
Upgraded from 14 to 16 early, but failed to upgrade to 18

**Comment by lewisdonofrio** [comment]
what error you getting?

**Comment by ksjogo** [comment]
Same error and logs for 18.04 to 18.10.

**Comment by lewisdonofrio** [comment]
What error, please get screenshot or paste error info here

**Comment by ksjogo** [comment]
No real error message is given:
```
 sudo -S env RELEASE_UPGRADER_NO_SCREEN=1 do-release-upgrade
[sudo] password for ksjogo:
Checking for a new Ubuntu release
Get:1 Upgrade tool signature [819 B]
Get:2 Upgrade tool [1,255 kB]
Fetched 1,256 kB in 0s (0 B/s)
authenticate 'cosmic.tar.gz' against 'cosmic.tar.gz.gpg'
extracting 'cosmic.tar.gz'

Reading cache
pcilib: Cannot open /proc/bus/pci
lspci: Cannot find any working access method.

Checking package manager
Reading package lists... Done
Building dependency tree
Reading state information... Done
Hit http://archive.ubuntu.com/ubuntu bionic InRelease
Get:1 http://security.ubuntu.com/ubuntu bionic-security InRelease [83.2 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
Get:3 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
Fetched 247 kB in 6s (40.2 kB/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done

Restoring original system state

Aborting
Reading package lists... Done
Building dependency tree
Reading state information... Done

```

```
2018-11-07 22:05:14,884 INFO Using config files '['./DistUpgrade.cfg']'
2018-11-07 22:05:14,884 INFO uname information: 'Linux LPT-PC0VR8VL 4.4.0-18267-Microsoft #1001-Microsoft Sat Oct 20 19:08:00 PST 2018 x86_64'
2018-11-07 22:05:15,323 INFO apt version: '1.6.6'
2018-11-07 22:05:15,323 INFO python version: '3.6.6 (default, Sep 12 2018, 18:26:19)
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]'
2018-11-07 22:05:15,332 INFO release-upgrader version '18.10.11.1' started
2018-11-07 22:05:15,361 INFO locale: 'en_GB' 'UTF-8'
2018-11-07 22:05:15,473 WARNING estimate_kernel_initrd_size_in_boot() returned '0' for kernel?
2018-11-07 22:05:15,473 WARNING estimate_kernel_initrd_size_in_boot() returned '0' for initrd?
2018-11-07 22:05:15,664 DEBUG Using 'DistUpgradeViewText' view
2018-11-07 22:05:15,742 DEBUG enable dpkg --force-overwrite
2018-11-07 22:05:15,791 DEBUG creating statefile: '/var/log/dist-upgrade/apt-clone_system_state.tar.gz'
2018-11-07 22:05:24,398 DEBUG lsb-release: 'bionic'
2018-11-07 22:05:24,399 DEBUG _pythonSymlinkCheck run
2018-11-07 22:05:24,401 DEBUG openCache()
2018-11-07 22:05:24,401 DEBUG No such plugin directory: ./plugins
2018-11-07 22:05:24,402 DEBUG plugins for condition 'PreCacheOpen' are '[]'
2018-11-07 22:05:24,402 DEBUG plugins for condition 'cosmicPreCacheOpen' are '[]'
2018-11-07 22:05:24,402 DEBUG plugins for condition 'from_bionicPreCacheOpen' are '[]'
2018-11-07 22:05:24,402 DEBUG quirks: running PreCacheOpen
2018-11-07 22:05:24,402 DEBUG running Quirks.PreCacheOpen
2018-11-07 22:05:25,256 DEBUG /openCache(), new cache size 61463
2018-11-07 22:05:25,257 DEBUG need_server_mode(): run in 'desktop' mode, (because of pkg 'ubuntu-desktop')
2018-11-07 22:05:25,257 DEBUG checkViewDepends()
2018-11-07 22:05:25,257 DEBUG running doUpdate() (showErrors=False)
2018-11-07 22:05:28,969 DEBUG openCache()
2018-11-07 22:05:29,759 DEBUG /openCache(), new cache size 61463
2018-11-07 22:05:29,760 DEBUG doPostInitialUpdate
2018-11-07 22:05:29,760 DEBUG plugins for condition 'PostInitialUpdate' are '[]'
2018-11-07 22:05:29,761 DEBUG plugins for condition 'cosmicPostInitialUpdate' are '[]'
2018-11-07 22:05:29,761 DEBUG plugins for condition 'from_bionicPostInitialUpdate' are '[]'
2018-11-07 22:05:29,761 DEBUG quirks: running cosmicPostInitialUpdate
2018-11-07 22:05:29,761 DEBUG running Quirks.cosmicPostInitialUpdate
2018-11-07 22:05:29,792 DEBUG abort called
2018-11-07 22:05:29,793 DEBUG openCache()
2018-11-07 22:05:30,639 DEBUG /openCache(), new cache size 61463
```

**Comment by CapinWinky** [comment]
Same problem, tried editing sources.list to replace xenial with bionic and do 

`sudo apt dist-upgrade`

but that also failed

I've also had to do 

`sudo chmod 777 /var/cache/app-info/xapian/default -R`

a lot for just regular apt commands and I don't know what that's about.

**Comment by jiceorange** [comment]
I'm seeing a similar problem. No errors, but the release upgrade stops shortly after starting and restores the orginal state.  

I'm on Win 10 v 1809:

`Microsoft Windows [Version 10.0.17763.292]`

When I use the command: sudo do-release-upgrade, I see this output:

```
Reading cache
pcilib: Cannot open /proc/bus/pci
lspci: Cannot find any working access method.
Checking package manager
Reading package lists... Done
Building dependency tree
Reading state information... Done
Hit http://security.ubuntu.com/ubuntu xenial-security InRelease
Hit http://archive.ubuntu.com/ubuntu xenial InRelease
Hit http://archive.ubuntu.com/ubuntu xenial-updates InRelease
Get:1 http://archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]
Fetched 107 kB in 6s (0 B/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done
Restoring original system state
Aborting
Reading package lists... Done
Building dependency tree
Reading state information... Done
=== Command detached from window (Tue Feb  5 12:30:50 2019) ===
```


And, my main.log looks similar to the original poster.

**Reply by LaurentRDC** [reply]
I've been able to progress on this by uninstalling the "screen":

```bash
sudo apt-get remove screen
```

**Comment by CapinWinky** [comment]
I ended up installing the Ubuntu metro app and starting from scratch. I had the old style install in the lxss folder and simply couldn't get it to work.

**Comment by jiceorange** [comment]
I've got some more info and hope that someone can provide a hint. Based on an above comment about some of us having problems with apt-get update problems with the permission of  /var/cache/app-info/xapian/default , I tried changing permissions before trying to update the release:

`sudo chmod -R a+rX,u+w /var/cache/app-info/xapian/defaut`

Now, I get a little further, but eventually run into the same errors with database permissions.
`** (appstreamcli:1726): CRITICAL **: Error while moving old database out of the way.  `

Has anyone discovered the root cause of the permission problems that some of us experience? Is this WSL specific?

```
Reading cache
pcilib: Cannot open /proc/bus/pci
lspci: Cannot find any working access method.
Checking package manager
Reading package lists... Done
Building dependency tree
Reading state information... Done
Hit http://archive.ubuntu.com/ubuntu xenial InRelease
Get:1 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [109 kB]
Get:2 http://security.ubuntu.com/ubuntu xenial-security InRelease [109 kB]
Get:3 http://archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]
Get:4 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages [911 kB]
Get:5 http://security.ubuntu.com/ubuntu xenial-security/main amd64 Packages [613 kB]
Get:6 http://archive.ubuntu.com/ubuntu xenial-updates/main Translation-en [367 kB]
Get:7 http://security.ubuntu.com/ubuntu xenial-security/main Translation-en [253 kB]
Get:8 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 DEP-11 Metadata [318 kB]
Get:9 http://security.ubuntu.com/ubuntu xenial-security/main amd64 DEP-11 Metadata [67.9 kB]
Get:10 http://security.ubuntu.com/ubuntu xenial-security/main DEP-11 64x64 Icons [67.1 kB]
Get:11 http://archive.ubuntu.com/ubuntu xenial-updates/main DEP-11 64x64 Icons [229 kB]
Get:12 http://security.ubuntu.com/ubuntu xenial-security/universe amd64 Packages [422 kB]
Get:13 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages [727 kB]
Get:14 http://security.ubuntu.com/ubuntu xenial-security/universe Translation-en [168 kB]
Get:15 http://security.ubuntu.com/ubuntu xenial-security/universe amd64 DEP-11 Metadata [116 kB]
Get:16 http://security.ubuntu.com/ubuntu xenial-security/universe DEP-11 64x64 Icons [173 kB]
Get:17 http://archive.ubuntu.com/ubuntu xenial-updates/universe Translation-en [300 kB]
Get:18 http://security.ubuntu.com/ubuntu xenial-security/multiverse amd64 DEP-11 Metadata [2,464 B]
Get:19 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 DEP-11 Metadata [252 kB]
Get:20 http://archive.ubuntu.com/ubuntu xenial-updates/universe DEP-11 64x64 Icons [352 kB]
Get:21 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse amd64 DEP-11 Metadata [5,956 B]
Get:22 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse DEP-11 64x64 Icons [14.3 kB]
Get:23 http://archive.ubuntu.com/ubuntu xenial-backports/main amd64 DEP-11 Metadata [3,328 B]
Get:24 http://archive.ubuntu.com/ubuntu xenial-backports/universe amd64 DEP-11 Metadata [5,100 B]
Fetched 5,693 kB in 6s (582 kB/s)
** (appstreamcli:1573): CRITICAL **: Error while moving old database out of the way.
AppStream cache update failed.
** (appstreamcli:1630): CRITICAL **: Error while moving old database out of the way.
AppStream cache update failed.
** (appstreamcli:1678): CRITICAL **: Error while moving old database out of the way.
AppStream cache update failed.
** (appstreamcli:1726): CRITICAL **: Error while moving old database out of the way.
AppStream cache update failed.
Reading package lists... Done
Building dependency tree
Reading state information... Done
Restoring original system state
Aborting
Reading package lists... Done
Building dependency tree
Reading state information... Done
=== Command detached from window (Wed Feb  6 14:22:07 2019) === ```

**Comment by rbalint** [comment]
@jiceorange It seems to be WSL specific: https://github.com/microsoft/WSL/issues/640#issuecomment-335546231

**Comment by jiceorange** [comment]
Thanks.  I saw that thread.  But didn't pay attention to that comment. Seems a while ago. No solution yet?

**Comment by PeeJay** [comment]
FYI I just had the same problem updating 19.04 to 19.10 on a bare metal Ubuntu install. I tracked the problem down to snappy being corrupted. Even apt reinstall didn't fix it, I had to purge it and install it. Now install progresses normally.

**Reply by horros** [reply]
This was my problem on WSL 2 trying to upgrade from Ubuntu 18.04 to 20.04. Running apt remove  --purge snapd fixed it for me.

**Reply by treysis** [reply]
Had the same problem on 20.10 upgrading to 21.04. Weirdly, two weeks ago upgrading from 20.04 worked perfectly fine, now even upgrading from 20.04 to 20.10 doesn't work anymore.

**Reply by cherepo** [reply]
Weird, remove snapd fixed it for me as well.

**Reply by zhoushun123** [reply]
The same problem on 20.10 upgrading to 21.04.Works for me now

**Reply by nayyaung** [reply]
The same thing happened from upgrading WSL Ubuntu 18 to 20. removing snapd fixed it for me.

**Reply by trjw** [reply]
+1

**Reply by Klozz** [reply]
it worked :D thnks

**Reply by pscoetsee** [reply]
Thanks a lot, solved it for me as well.

**Reply by TheLinx79** [reply]
Thanks - my problem from 19.04 to 21.04 as well. But how on earth did you figure out that it was Snapd?

**Reply by ghost** [reply]
As noted in my comment further down, there is a log file at `/var/log/dist-upgrade/main.log` that probably has a hint of `snapd` causing issues (although I ran into `ubuntu-desktop` causing issues instead)

**Reply by ebelfer2020** [reply]
Thanks,  removing snapd worked.

**Reply by hakanai** [reply]
Fixed for me too. You'd hope they'd actually test the upgrade tool before release, but oh well.

**Reply by Mischala** [reply]
Can confirm removing `snapd`  allowed me to upgrade from 20.04 to 22.04
not sure what `snapd` is up to, but it sure doesn't seem to like WSL2

**Reply by Dzonixy** [reply]
Worked for me too. Thanks!

**Reply by AleksandrHovhannisyan** [reply]
Can also confirm this is still working. I was on Ubuntu 18.04 and this helped me upgrade to 20. Thanks!

**Comment by therealkenc** [comment]
This one was a general Linux discussion a xenial upgrade rollback, which can happen for reasons, for which there isn't going to be any repro in this thread. **Unrelated** #640 became quiet many moons ago without any convincing explanation forthcoming as to what got better, but take the win.

**Comment by ghost** [comment]
Hope this helps someone else.

There may be additional info in `/var/log/dist-upgrade/main.log` after the failure.  I found this in mine:
```DEBUG need_server_mode(): run in 'desktop' mode, (because of pkg 'ubuntu-desktop')```

After removing ubuntu-desktop, the upgrade working smoothly, despite still getting the errors about /proc/bus/pci and lspci missing.

Don't ask why I had ubuntu-desktop installed on WSL in the first place.

**Reply by jiceorange** [reply]
Interesting. I too had installed ubuntu-desktop (I was experimenting to see how much of the ubuntu experience I could get running). And, I see the same DEBUG message (but it's only a debug, so I am surprised that would trigger a rollback). 

I have long since taken the plunge and went ahead and installed the Ubuntu 18.04 App (so, that would imply I need to install a different app when I am ready to upgrade again). After that, I installed the packages that I wanted from the Ubuntu app and then used the hooks to the Windows file system to copy over my home directory and other files that I wanted to migrate. As WSL encourages semi-disposable linux environments, I should probably figure out a way to script an automatic setup of a useful environment. But, up until now, have mostly been "playing" and evaluating what is missing or doesn't yet work for me. I haven't taken the WSL2 plunge yet.

**Reply by joelmgallant** [reply]
It definitely helped me, thanks!
I had the same issue - `do-release-upgrade` failing on WSL2 Ubuntu, with `ubuntu-desktop` package installed.
Was just receiving a message of "Aborted" during the upgrade - removing `ubuntu-desktop` allows it to proceed!

**Reply by santoshpuranik** [reply]
Thanks! this helped me too! Wonder what goes wrong when ubuntu-desktop is installed, a more meaningful error message would be nice.

**Reply by dacid44** [reply]
I just had this same problem, removing `ubuntu-desktop` seems to have fixed it (upgrading from 18.04 to 20.04). As for why I don't know anything for certain, but I've had weirdness before with desktop environments requiring systemd (which isn't really a thing in WSL). That would be my guess but it's not really based on anything in particular.

**Reply by rcarmonad** [reply]
I have this issue, but run this https://github.com/microsoft/WSL/discussions/3489#discussioncomment-673231 and works the upgrade

**Reply by ojnickel** [reply]
where should i remove ubuntu-desktop?

**Reply by lewisdonofrio** [reply]
terminal?  apt-get remove ubuntu-desktop (but if your upgrading you might try to remove snapd and see if that helps?)

**Comment by lewisdonofrio** [comment]
I've used full xfce4 gui for almost 4 years now on my work machines and my daily driver....I have a strange "no ms store" corp locked down w10 install so I use command line to skip past doman group policy lockdowns...
https://tinyurl.com/donofriovines/photos/tech-2018/wsl1-and-windows10

**Comment by treysis** [comment]
I have the same problem trying to upgrade 20.04 to 20.10. None of the proposed solutions here worked. Weirdly I was able to upgrade another WSL1 Ubuntu install last weekend from 20.04 to 20.10 without problems.

**Reply by StKob** [reply]
I found on reddit that snapd package can result in do-release-upgrade termination. After removing it everything works as it should.

**Reply by treysis** [reply]
Yeah, me too. Just forgot to update this post. Snapd was the culprit.

**Reply by pouyaSamie** [reply]
> I found on reddit that snapd package can result in do-release-upgrade termination. After removing it everything works as it should.

that fixed my problem on wsl2. thanx

**Reply by calebharris** [reply]
Ditto here, trying to upgrade Ubuntu 20.04 LTS using `do-release-upgrade` installed on WSL 2.

**Reply by therealkenc** [reply]
There are some suggestions in #6942.

**Reply by pramodhrachuri** [reply]
> I found on reddit that snapd package can result in do-release-upgrade termination. After removing it everything works as it should.

Solved my problem. Thanks!

**Reply by martin-jorgensen** [reply]
`sudo apt remove snapd` solved the problem for me too

**Reply by slix** [reply]
Removing `snapd` worked for me on WSL1.

**Reply by grufield** [reply]
Removing snapd worked for me on WSL2 ubuntu 18.04

**Reply by SoniCoder** [reply]
removing snapd worked

**Comment by ojnickel** [comment]
i did b oth and they worked! thx!!

Am Fr., 11. Juni 2021 um 16:32 Uhr schrieb Lewis L. Donofrio <
***@***.***>:

**Comment by uranusx86** [comment]
after remove `screen` and `snapd` package, I have successfully upgraded ubuntu 16 to 18.

**Reply by Moekr** [reply]
this is useful for me

**Reply by bastafidli** [reply]
This has fixed it for me as well, when upgrading from xenial to bionic

sudo apt-get remove screen snapd

**Reply by Ryan-Dia** [reply]
oh my god!! thank!!!!! i love

**Reply by IGR2014** [reply]
YES, you did the trick! Many thanks! Helped me on my way from 20.04 up!

**Reply by lpd6375** [reply]
Thanks dude. It works for me. Upgrade to Ubuntu 21.04

**Reply by ggbong734** [reply]
Running `sudo apt-get remove screen snapd` and then `sudo do-release-upgrade` worked for me too. Upgraded my Ubuntu from 18.04 to the latest in WSL.

**Reply by GoncaloCoutinho** [reply]
This worked for me too! Thank you very much!

**Reply by noobmagnet** [reply]
> after remove `screen` and `snapd` package, I have successfully upgraded ubuntu 16 to 18.

How did you figure this out? Mine is now working finally after removing only snapd so thank you for the solution!

**Reply by BacchusPlateau** [reply]
Thank you for this...could not get it to work until I stumbled upon this post.

**Reply by kczx3** [reply]
Does anyone actually know why removing of these is required?

**Comment by jjakob** [comment]
Had the same problem when trying to release-upgrade xenial to bionic on a bare metal Ubuntu install, not WSL. Purging snapd fixed it.

**Comment by GeoffreyEmerson** [comment]
Removing `snapd` caused some unexpected pain.

```
% sudo apt install snapd
Reading package lists... Done
Building dependency tree
Reading state information... Done
Suggested packages:
  zenity | kdialog
E: Could not read response to hello message from hook [ ! -f /usr/bin/snap ] || /usr/bin/snap advise-snap --from-apt 2>/dev/null || true: Success
E: Could not read response to hello message from hook [ ! -f /usr/bin/snap ] || /usr/bin/snap advise-snap --from-apt 2>/dev/null || true: Success
%
```

Some apt commands break like this before doing anything.

After messing around trying to fix snapd and rebooting, for some reason the update command seems to now work.

**Reply by ghost** [reply]
See https://github.com/microsoft/WSL/issues/4640 for fixes.

**Reply by GeoffreyEmerson** [reply]
Luckily the upgrade process fixed it, but thanks.
